bundles/io/org.openhab.io.myopenhab/src/main/java/org/openhab/io/myopenhab/internal/MyOHClient.java

```diff
@@ -0,0 +1,559 @@
+	 */
+
+	public void onEvent(String event, JSONObject data) {
+		// TODO Auto-generated method stub
+		logger.debug("on(): " + event);
+		if (event.equals("request")) {
+			try {
+				// Get myOH uniq request Id
+				int requestId = data.getInt("id");
+				logger.debug(String.format("Got request %d", requestId));
+				// Get request path
+				String requestPath = data.getString("path");
+				// Get request method
+				String requestMethod = data.getString("method");
+				// Get request body
+				String requestBody = data.getString("body");
+				// Get JSONObject for request headers
+				JSONObject requestHeadersJson = data.getJSONObject("headers");
+				logger.debug(requestHeadersJson.toString());
+				// Get JSONObject for request query parameters
+				JSONObject requestQueryJson = data.getJSONObject("query");
+				Iterator<String> headersIterator = requestHeadersJson.keys();
+				// Create URI builder with base request URI of openHAB and path from request
+				String newPath = URIUtil.addPaths(mOHBaseUrl, requestPath);
+				Iterator<String> queryIterator = requestQueryJson.keys();
+				// Add query parameters to URI builder, if any
+				newPath +="?";
+				while (queryIterator.hasNext()) {
+					String queryName = queryIterator.next();
+					newPath += queryName;
+					newPath += "=";
+					newPath += URLEncoder.encode(requestQueryJson.getString(queryName), "UTF-8");
+					if (queryIterator.hasNext())
+						newPath += "&";
+				}
+				// Finally get the future request URI
+				URI requestUri = new URI(newPath);
+				// All preparations which are common for different methods are done
+				// Now perform the request to openHAB
+				// If method is GET
+				logger.debug("Request method is " + requestMethod);
+				MyOHExchange exchange = new MyOHExchange(requestId);
+				exchange.setURI(requestUri);
+				exchange.setRequestHeaders(requestHeadersJson);
+				exchange.setRequestHeader("X-Forwarded-Proto", "https");
+				if (requestMethod.equals("GET")) {
+					exchange.setMethod("GET");
+				} else if (requestMethod.equals("POST")) {
+					exchange.setMethod("POST");
+					Buffer requestContent = new ByteArrayBuffer(requestBody);
+					exchange.setRequestContent(requestContent);
+				} else if (requestMethod.equals("PUT")) {
+					exchange.setMethod("PUT");
+					Buffer requestContent = new ByteArrayBuffer(requestBody);
+					exchange.setRequestContent(requestContent);
+				} else {
+					// TODO: Reject unsupported methods
+					logger.error("Unsupported request method " + requestMethod);
+					return;
+				}
+				mJettyClient.send(exchange);
+				// If successfully submitted request to http client, add it to the list of currently
+				// running requests to be able to cancel it if needed
+				mRunningRequests.put(requestId, exchange);
+			} catch (JSONException e) {
+				logger.error(e.getMessage());
+			} catch (IOException e) {
+				logger.error(e.getMessage());
+			} catch (URISyntaxException e) {
+				logger.error(e.getMessage());
+			}
```

Comment:
extract into handleRequestEvent() method (same applies to the subsequent else blocks)